pkg/security/security_profile/activity_tree/activity_tree.go

```diff
@@ -919,13 +919,13 @@ func (at *ActivityTree) ExtractPaths() (map[string][]string, map[string][]string
 }
 
 // ExtractSyscalls return the syscalls present in an activity tree
-func (at *ActivityTree) ExtractSyscalls() []string {
+func (at *ActivityTree) ExtractSyscalls(arch string) []string {
 	var syscalls []string
 
 	at.visit(func(processNode *ProcessNode) {
 		for _, s := range processNode.Syscalls {
-			syscall := model.Syscall(s).ConvertSyscallName()
-			syscalls = append(syscalls, syscall)
+			sycallKey := utils.SyscallKey{Arch: arch, ID: s}
+			syscalls = append(syscalls, utils.Syscalls[sycallKey])
```

Comment:
IMHO we should check the return of utils.Syscalls[sycallKey] in case the requested syscalls isn't present (which could happen since the dict only contains native syscalls ABI IDs)